code/modules/power/apc.dm

```diff
@@ -1287,6 +1289,21 @@ obj/machinery/power/apc/proc/autoset(var/cur_state, var/on)
 	update_icon()
 	return 1
 
+/obj/machinery/power/apc/proc/set_light_color(var/color) //just changes the colors of the ship's lighting.
```

Comment:
Personally, I would have this proc take the `/decl/security_level` itself as an argument. Then, checking for nulls to reset, have `lightmode` just set to `security_level_lightmode`.